### PR TITLE
[Console] Remove redundant method getSaturation()

### PR DESCRIPTION
**Pull request by ivanpepelko**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | none
| License       | MIT
| Doc PR        | none

1. in `getSaturation()`, `$diff` is compared against `int` zero. This will only yield `true` when RGB values contain min/max color value (`0` or `255`). It will be always `false` for floats.
2. in `degradeHexColorToAnsi4()` value returned from `getSaturation()` is `round`ed, which returns float, thus making comparison to `int` zero useless.
3. in the same method, bitwise operations (final `return` statement) will convert number to `int` anyway, so the explicit cast was removed

In conclusion, `getSaturation` is redundant. Tests are expanded to show that even in min/max cases everything works fine.

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Review by maxbeckers — APPROVED** [review]
LGTM

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Merging in 6.3 as this is not a bug fix, but mainly an improvement.

**Comment by fabpot** [comment]
Thank you @ivanpepelko.

**Comment by downhiller** [comment]
I think the main issue was a bug on line 167:

    return (int) $diff * 100 / $v;

should have been

    return (int) ($diff * 100 / $v);

$diff was always less between 0.0 and 1.0, so would always (int) to 0, or 1.

My contribution feels a bit redundant given that they've moved on and refactored a while back\, but might help someone wondering why getSaturation doesn't return anything useful.
